vars/essentialsTest.groovy

```diff
@@ -0,0 +1,41 @@
+def call(Map params = [:]) {
+    def baseDir = params.containsKey('baseDir') ? params.baseDir : "."
+    def metadataFile = params.containsKey('metadataFile') ? params.metadataFile : "essentials.yml"
+    def labels = params.containsKey('labels') ? params.labels : "docker && highmem"
+    cdkata(metadataFile, labels, baseDir)
+
+    node(labels) {
+        stage("Checkout") {
+            infra.checkout()
+        }
+
+        dir(baseDir) {
+            def metadataPath = "${pwd()}/${metadataFile}"
+            metadata = readYaml(file: metadataPath)
+
+            def customBOM = "${pwd tmp: true}/custom.bom.yml"
+            def customWAR = "${pwd tmp: true}/custom.war"
+            def customWarURI = "file://" + customWAR
```

Comment:
Unsafe URL construction. You need to URL-escape the file path at a minimum. Really we ought to have a step in `pipeline-utility-steps` to do this I guess: convert a (perhaps workspace-relative) file path to a URI using `Path.toUri`.